x-pack/plugins/apm/public/components/app/service_overview/service_overview_transactions_table/index.tsx

```diff
@@ -258,6 +195,127 @@ export function ServiceOverviewTransactionsTable(props: Props) {
       },
     },
   ];
+}
+
+export function ServiceOverviewTransactionsTable(props: Props) {
+  const { serviceName } = props;
+  const { transactionType } = useApmServiceContext();
+  const {
+    uiFilters,
+    urlParams: { start, end, latencyAggregationType },
+  } = useUrlParams();
+
+  const [tableOptions, setTableOptions] = useState<{
+    pageIndex: number;
+    sort: {
+      direction: SortDirection;
+      field: SortField;
+    };
+  }>({
+    pageIndex: 0,
+    sort: DEFAULT_SORT,
+  });
+
+  const {
+    data = {
+      totalItemCount: 0,
+      items: [],
+      tableOptions: {
+        pageIndex: 0,
+        sort: DEFAULT_SORT,
+      },
+    },
+    status,
+  } = useFetcher(() => {
+    if (!start || !end || !latencyAggregationType || !transactionType) {
+      return;
+    }
+
+    return callApmApi({
+      endpoint:
+        'GET /api/apm/services/{serviceName}/transactions/groups/overview',
+      params: {
+        path: { serviceName },
+        query: {
+          start,
+          end,
+          uiFilters: JSON.stringify(uiFilters),
+          numBuckets: 20,
+          sortField: tableOptions.sort.field,
+          sortDirection: tableOptions.sort.direction,
+          transactionType,
+          latencyAggregationType: latencyAggregationType as LatencyAggregationType,
+        },
+      },
+    }).then((response) => {
+      return {
+        items: response.transactionGroups,
+        totalItemCount: response.totalTransactionGroups,
+        tableOptions: {
+          pageIndex: tableOptions.pageIndex,
+          sort: {
+            field: tableOptions.sort.field,
+            direction: tableOptions.sort.direction,
+          },
```

Comment:
I'm not sure we need all of this still, as this was mostly in place to accommodate for the fact that for every page/sort change we called the API again. This table maybe should be an EuiInMemoryTable.

Reply:
I removed it and changed it to use `ManagedTable` component.